src/main/java/org/cbio/gdcpipeline/step/ClinicalDataStep.java

```diff
@@ -0,0 +1,160 @@
+package org.cbio.gdcpipeline.step;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.cbio.gdcpipeline.model.cbio.CBioClinicalDataModel;
+import org.cbio.gdcpipeline.processor.ClinicalDataProcessor;
+import org.cbio.gdcpipeline.reader.ClinicalStreamReader;
+import org.cbio.gdcpipeline.writer.ClinicalPatientDataWriter;
+import org.cbio.gdcpipeline.writer.ClinicalSampleDataWriter;
+import org.springframework.batch.core.Step;
+import org.springframework.batch.core.configuration.annotation.EnableBatchProcessing;
+import org.springframework.batch.core.configuration.annotation.JobBuilderFactory;
+import org.springframework.batch.core.configuration.annotation.StepBuilderFactory;
+import org.springframework.batch.core.configuration.annotation.StepScope;
+import org.springframework.batch.item.ItemWriter;
+import org.springframework.batch.item.file.MultiResourceItemReader;
+import org.springframework.batch.item.support.CompositeItemWriter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Created by Dixit on 16/06/17.
+ */
+@EnableBatchProcessing
+@Configuration
+public class ClinicalDataStep {
+
+    @Autowired
+    StepBuilderFactory stepBuilderFactory;
+
+    @Autowired
+    JobBuilderFactory jobBuilderFactory;
+
+
+    private static Log LOG = LogFactory.getLog(ClinicalDataStep.class);
+
+    @Value("${chunk.interval}")
+    private int chunkInterval;
+
+    //TODO Resolve Bean issue
+    // Spring cannot get @Value from execution context's since this class is marked with @Configuration
+    // and they are initialized only after AppContext is run
+    // Cannot mark this step as step scope or jobscope as well : null pointer exception
+    // possibility : can retrieve values from listener but how to pass to multi resource reader ?
+    // configure spring job launcher code ?
+
+    //@Value("#{jobParameters[sourceDirectory]}")
```

Comment:
You shouldn't need any of these parameters in the Step class, the Step should just be a recipe for the reader/processor/writer and any other spring components that need to work on the data.